lib/rgb_matrix/animation/config/field_type.ex

```diff
@@ -7,6 +7,20 @@ defmodule RGBMatrix.Animation.Config.FieldType do
           __MODULE__.Integer.t()
           | __MODULE__.Option.t()
 
-  @callback validate(t, any) :: :ok | :error
-  @callback cast(t, any) :: {:ok, any} | :error
+  @typedoc """
+  `field` is a union of all the `FieldType.<field_type>` modules
+  """
+  @type field ::
+          __MODULE__.Integer
+          | __MODULE__.Option
+
+  @typedoc """
+  `value` is a union of all valid field value types
+  """
+  @type value ::
+          __MODULE__.Integer.value()
+          | __MODULE__.Option.value()
```

Comment:
I could move the union out to the module(s) which might use the `FieldType.field`/`.value` types, but I'm trying to ease some clutter in the type specs by putting it closer to the source and describing it once. Maybe I'm abstracting a bit much and the naming of types should be closer to where they get used, but then there are more types to remember to update.

I'm trying to describe a type of "All kinds of fields that can be defined" and "All kinds of values for fields". I'm open to suggestions for names.

We could maybe rename `FieldType` -> `Field`, which reads a little better, and then the union types could become `Field.any_type` and `Field.any_value`.

Reply:
>We could maybe rename `FieldType` -> `Field`, which reads a little better, and then the union types could become `Field.any_type` and `Field.any_value`.

I like the sound of where that's headed. It's certainly clearer when reading it. Along those lines, is `Field.any_value` just `any`?

Reply:
Hhhmmmm... I *think* it would be appropriate for `Field.any_type` to become `Field.any`, and `any_value` could become `Field.value`, which definitely reads as "a value for a field".

@doughsay What do you think about this particular idea?

Reply:
Sorry, I didn't mean `Field.any`, I meant plain old built-in `any` (i.e. `term`).